contrib/bloop/src/mill/contrib/bloop/BloopImpl.scala

```diff
@@ -405,6 +330,81 @@ class BloopImpl(ev: () => Evaluator, wd: Path) extends ExternalModule { outer =>
 
+    ////////////////////////////////////////////////////////////////////////////
+    // Platform (Jvm/Js/Native)
+    ////////////////////////////////////////////////////////////////////////////
+
+    def jsLinkerMode(m: JavaModule): Task[Config.LinkerMode] =
+      (m.asBloop match {
+        case Some(bm) => T.task(bm.linkerMode())
+        case None     => T.task(None)
+      }).map(_.getOrElse(Config.LinkerMode.Debug))
+
+    def platform: Task[BloopConfig.Platform] = module match {
+      case m: ScalaJSModule =>
+        T.task {
+          BloopConfig.Platform.Js(
+            BloopConfig.JsConfig.empty.copy(
+              version = m.scalaJSVersion(),
+              mode = jsLinkerMode(m)(),
+              kind = m.moduleKind() match {
+                case ModuleKind.NoModule => Config.ModuleKindJS.NoModule
+                case ModuleKind.CommonJSModule =>
+                  Config.ModuleKindJS.CommonJSModule
+                case ModuleKind.ESModule => Config.ModuleKindJS.ESModule
+              },
+              emitSourceMaps = m.jsEnvConfig() match {
+                case c: JsEnvConfig.NodeJs => c.sourceMap
+                case _                     => false
+              },
+              jsdom = Some(false)
+            ),
+            mainClass = module.mainClass()
+          )
+        }
+      case m: ScalaNativeModule =>
+        T.task {
+          BloopConfig.Platform.Native(
+            BloopConfig.NativeConfig.empty.copy(
+              version = m.scalaNativeVersion(),
+              mode = m.releaseMode() match {
+                case ReleaseMode.Debug       => BloopConfig.LinkerMode.Debug
+                case ReleaseMode.ReleaseFast => BloopConfig.LinkerMode.Release
+                case ReleaseMode.ReleaseFull => BloopConfig.LinkerMode.Release
+              },
+              gc = m.nativeGC(),
+              targetTriple = m.nativeTarget(),
+              clang = m.nativeClang().toNIO,
+              clangpp = m.nativeClangPP().toNIO,
+              options = Config.NativeOptions(
+                m.nativeLinkingOptions().toList,
+                m.nativeCompileOptions().toList
+              ),
+              linkStubs = m.nativeLinkStubs()
+            ),
+            mainClass = module.mainClass()
+          )
+        }
+      case _ =>
+        T.task {
+          BloopConfig.Platform.Jvm(
+            BloopConfig.JvmConfig(
+              home = T.env.get("JAVA_HOME").map(s => Path(s).toNIO),
+              options = {
+                // See https://github.com/scalacenter/bloop/issues/1167
+                val forkArgs = module.forkArgs().toList
+                if (forkArgs.exists(_.startsWith("-Duser.dir="))) forkArgs
+                else s"-Duser.dir=$wd" :: forkArgs
+              }
+            ),
+            mainClass = module.mainClass(),
+            runtimeConfig = None,
+            classpath = Some(classpath().map(_.toNIO).toList),
```

Comment:
This is the only line that changed. 

It would be good to understand what bloop uses this for

Reply:
Isn't that just classpath for compilation?

Reply:
Nope, the classpath for compilation is [there](https://github.com/com-lihaoyi/mill/blob/abe5220d8958973a9afdc058d96df387f8f02520/contrib/bloop/src/mill/contrib/bloop/BloopImpl.scala#L430)

Reply:
I mean, yeah, the value that I'm assigning to the config (in this PR) is the the same as the one that's used for compilation, but here I'm asking about the meaning of that particular configuration element.

Reply:
Wait, that is in platform, let me check it.

Reply:
So this was added as a separate configuration for running on JVM. More info in the commit https://github.com/scalacenter/bloop/commit/8898a93c0f88219a2edbe53b6f8d0765b58d20f7

Reply:
Right so according to the comment it could be left empty. Right ?

Reply:
It will use the main classpath if it's missing. Looks like it.